sdk/common.go

```diff
@@ -18,11 +18,17 @@ import (
 	"github.com/edgelesssys/contrast/internal/manifest"
 )
 
+// validatorsFromManifestOverride is used to override the validators in tests.
+var validatorsFromManifestOverride func(*certcache.CachedHTTPSGetter, *manifest.Manifest, *slog.Logger) ([]atls.Validator, error)
+
 // ValidatorsFromManifest returns a list of validators corresponding to the reference values in the given manifest.
 // Originally an unexported function in the contrast CLI.
 // Can be made unexported again, if we decide to move all userapi calls from the CLI to the SDK.
 // Validators MUST NOT be used concurrently.
 func ValidatorsFromManifest(kdsGetter *certcache.CachedHTTPSGetter, m *manifest.Manifest, log *slog.Logger) ([]atls.Validator, error) {
+	if validatorsFromManifestOverride != nil {
+		return validatorsFromManifestOverride(kdsGetter, m, log)
+	}
```

Comment:
Would be cleaner to transform the function into a method to perform the injection.

Reply:
I guess I was confused by "transform the function into a method". Is that what you mean?

```go
type Client struct {
  validatorGetter func(kdsGetter *certcache.CachedHTTPSGetter, m *manifest.Manifest, log *slog.Logger)
}

func New() Client {
  return &Client{validatorGetter: ValidatorsFromManifest}
}
```

If not, I'd appreciate if you could propose a change.